### PR TITLE
Add packages/streams (port of stream-staging-area experiment)

### PR DESCRIPTION
**Pull request by jonastemplestein**
## What

Moves the **stream staging-area** experiment from `cf-experiments` into this monorepo as a new package, `packages/streams` — byte-for-byte the same code except where it had to change to conform to iterate conventions and wire into the workspace. Deployment stays **wrangler-based (no alchemy)**, per request.

Contents (unchanged from source): the stream processor abstraction, the `Stream` + `StreamProcessorRunner` Durable Objects, CapnWeb-over-WebSocket RPC, browser/Node/Workers clients, and a TanStack Start `example-app/` worker with vitest + Playwright e2e suites.

## How it was brought in

- **Copied** the tree (`rsync`, minus `node_modules` and gitignored artifacts).
- **Vendored** the only cross-package dependency — the three `@cf-experiments/shared` modules it imported (`event`, `stream-processors`, `rpc-target`, which depend only on `zod` + `cloudflare:workers`) — into `src/shared/`, and repointed all 25 import sites. (`@iterate-com/shared`'s own `stream-processors` is a different implementation, so reuse would have been wrong.)
- **Style conformance:** `.ts` relative-import extensions (114 rewrites), PascalCase zod schema names, removed stale cf-experiments `eslint-disable` directives, fixed an `exhaustive-deps` warning and an ASI-hazard. `oxlint --deny-warnings` and `oxfmt` clean.
- **Workspace wiring:** renamed to `@iterate-com/streams` / `@iterate-com/streams-example-app`, dropped the `@cf-experiments/shared` dep, pointed Cloudflare deps at `catalog:cloudflare`, made the tsconfigs standalone (the cf-experiments base didn't exist here), and added a `packages/*/example-app` workspace glob so the nested app is a member.
- **cloudflared e2e determinism:** the worker dials an `external-url` subscriber exactly once (reconcile only re-dials on topology change / RPC break), and a fresh quick tunnel 530s for a few seconds after the URL prints. The test now waits for cloudflared to register an edge connection (+ propagation buffer) before configuring the subscription, so that single dial lands on a routable tunnel.

## Verification

| Suite | local miniflare | deployed worker |
|---|---|---|
| lib unit (`vitest`, 26) | ✅ | — |
| e2e vitest (capnweb/processor) | ✅ 13 (+1 skip) | ✅ 13 (+1 skip) |
| cloudflared-gated e2e | n/a¹ | ✅ (stable 3/3) |
| Playwright (27) | ✅ | ✅ |

Deployed to `https://stream-staging-area.iterate-dev-preview.workers.dev` (account `376ef7…`, "iterate (dev/preview)") via `wrangler deploy`.

¹ The cloudflared test exercises a worker dialing **out** to a dev machine through a public quick tunnel — meaningful only for a remote worker. Against local miniflare the equivalent direct-`127.0.0.1` path is covered by the `localWorkerE2eIt` test (which passes); the localhost variant is correctly skipped against the deployed worker.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Large new surface area (Durable Objects, subscriptions, browser SQLite mirror) with extensive e2e coverage but limited prior integration in this repo; outbound dial/replay edge cases were explicitly fixed in tests.
> 
> **Overview**
> Introduces **`packages/streams`** (`@iterate-com/streams`) by porting the **stream-staging-area** experiment into the monorepo: stream processor contracts/runner, **Stream** and **StreamProcessorRunner** Durable Objects, CapnWeb WebSocket clients (browser/Node/Workers), and a TanStack Start **`example-app`** worker with vitest + Playwright e2e.
> 
> **Integration work:** renames packages for iterate, wires Cloudflare deps via **`catalog:cloudflare`**, vendors former **`@cf-experiments/shared`** pieces into **`src/shared/`** (instead of **`@iterate-com/shared`** stream code), and applies repo style (relative `.ts` import extensions, lint/format fixes). Deployment stays **Wrangler**-based through the example app.
> 
> **Behavior/tests called out in the port:** broad e2e coverage (capnweb append/replay/subscriptions, outbound processors, one-way batch delivery wire shape, browser OPFS SQLite mirror, multi-tab/split-view leadership), plus a **cloudflared** external-subscriber test made more deterministic by waiting for tunnel registration before **`subscription-configured`** so the worker’s single outbound dial hits a routable URL.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 272a92005d43bd120de521851a81a3d2b07300db. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/streams/src/workers/durable-objects/stream.ts [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+          streamMaxOffset: streamMaxOffsetBeforeHandshake,
+          subscriptionConfiguredEvent: configured.latestConfiguredEvent,
+          streamRuntimeState: this.runtimeState(),
+        });
+
+        this.#openConnection({
+          ...request,
+          direction: "outbound",
+          replayAfterOffset: request.replayAfterOffset ?? streamMaxOffsetBeforeHandshake,
+          subscriptionKey,
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply packages/streams/src/workers/durable-objects/stream.ts:688]: Fixed in the way you suggested: the outbound default `replayAfterOffset` now falls back to the subscription-configured event's own offset (`configured.latestConfiguredEvent.offset`) instead of `streamMaxOffset` at dial time.

Since `replayAfterOffset` is exclusive, this starts delivery at the first event *after* the subscription was added, so any events committed in the gap between the `subscription-configured` append and a successful dial are now replayed rather than skipped. Subscribers that return their own `replayAfterOffset` (e.g. a built-in runner resuming from a checkpoint) still win.
